leader-election-impl/src/main/java/com/palantir/paxos/PaxosResponsesWithRemote.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.paxos;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import org.immutables.value.Value;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.common.streams.KeyedStream;
+
+@Value.Immutable
+public abstract class PaxosResponsesWithRemote<S, T extends PaxosResponse> {
```

Comment:
Hmm, I'd prefer we use `REMOTE` or something rather than `S`, also possibly have the params the other way around (the thing extending PaxosResponse first).

Reply:
`S` initially stood for service, but checkstyle was complaining, not sure why I'm able to do it elsewhere.

Disagree about the second point, i.e. you get a response `T` by executing a command against `S(ERVICE)` similar to `method(self, args)` in python, `method.invoke(object, args...)` in java reflection etc.